### PR TITLE
Add `qjsc -r` flag to output raw bytecode

### PR DESCRIPTION
**Pull request by TooTallNate**
I want to allow users to ship a raw `.jsc` bytecode file for faster bootup times.

### CODE REVIEW

**Comment by saghul** [comment]
How will you handle handle bytecode incompatibility?

Edit, hit enter too fast. Bytecode generated by a given version of QJSC need not be compatible with previous or further versions of the engine. There is a BC version we bump when we make changes to the way bytecode is (de)serialized.

I support the approach, but I worry people are going to run into problems because this incompatibility is not obvious...

**Comment by TooTallNate** [comment]
> How will you handle handle bytecode incompatibility?

QuickJS already throws a syntax error, so on the technical level I'm not sure what else would need to be done: https://github.com/quickjs-ng/quickjs/blob/07fa1cbc4a8938085a716051c0f14a3453a885fb/quickjs.c#L35116-L35120

Socially, I plan on shipping an emscripten-compiled version of `qjsc` with my runtime, so the user code would always be compiled against the same quickjs version as the runtime itself. I suppose implementors might want to come up with their own approaches.

**Comment by saghul** [comment]
Sounds reasonable. We can also look into the syntax error message and make it more helpful if need be as well.

File: qjsc.c [hunk 1]
```diff
@@ -450,25 +462,29 @@ int main(int argc, char **argv)
     ctx = JS_NewContext(rt);
 
     /* loader for ES6 modules */
-    JS_SetModuleLoaderFunc(rt, NULL, jsc_module_loader, NULL);
+    OutputTypeEnum *output_type_ptr = (OutputTypeEnum *)js_malloc(ctx, sizeof(OutputTypeEnum));
```
Comment (saghul) [inline qjsc.c]: Why don't you make the output type a static variable at the top of the file and simplify this? There is already precedent with things like `outfile` and `strip` and then you don't need to shove it in there

Comment (TooTallNate) [reply qjsc.c]: 02dfbac0a77ed5ba98a5f7269814ce7d34701c58

**Review by saghul — APPROVED** [review]

**Comment by chqrlie** [comment]
There is a problem: the output file must be open in binary mode to output the raw bytecode:
```
    if (output_type == OUTPUT_RAW)
        fo = fopen(cfilename, "wb");
    else
        fo = fopen(cfilename, "w");
```
This is necessary for legacy architectures with end-of-line translation and other obscure text mode conversion schemes.
I would have also preferred `-b` (`--bytecode`) over `-r` for this option.

**Pull request merged**